ksql-cli/src/main/java/io/confluent/ksql/util/TabularRow.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.rest.entity.FieldInfo;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class TabularRow {
+
+  private final int width;
+  private final List<String> value;
+  private final List<String> header;
+
+  public TabularRow(
```

Comment:
`header` and `value` are used mutually exclusively.   They both map to a list of strings.  So this class could be simplified to just take the width and single `List<String>` param, pushing up the computation of _what_ is passed as the list of strings to the caller, or maybe two factory methods.

Reply:
the reason I have it this way is that in the near future I want to change the column format based on the header - so passing in both will soon be helpful (e.g. fixed width for int columns, etc...)

Reply:
OK, in that case I'd still go for two factory methods to make this clear:

`TabularRow.createHeader(headers)` vs `TabularRow.createRow(headers, row)`.

And you can still decouple this class from `FieldInfo` and `GenericRow` by having the calling code do the `toString` stuff.  Less couple is good!  Plus, you can process the headers once for the query, rather than per-row.

Reply:
> And you can still decouple this class from FieldInfo and GenericRow by having the calling code do the toString stuff.

I don't think the _current_ calling class would be a good place to do that. I could introduce another wrapper class (e.g. `Tabular`) that handled all of the row widths based on the field infos and things like that, but that is overkill for this PR. As it stands, it's pretty decoupled - it's just in a constructor (moved to be static initializer).